rollouts/controllers/remoterootsync_controller.go

```diff
@@ -121,10 +129,16 @@ func (r *RemoteRootSyncReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		// The object is being deleted
 		if controllerutil.ContainsFinalizer(&remoterootsync, myFinalizerName) {
 			// our finalizer is present, so lets handle any external dependency
-			if meta.IsStatusConditionTrue(remoterootsync.Status.Conditions, externalSyncCreatedConditionType) {
+			if r.isExternalSyncCreated(&remoterootsync) {
```

Comment:
Do we need to make this check here? I would expect us to just make the call to clean up external resources regardless, and it will just check that any external resources are not there. Whenever possible, we should just look at the actual state and determine what do to, unless it is prohibitively expensive.

Reply:
This check was introduced to allow us to recover, as in allowing a remote root sync to be successfully deleted, for a misconfigured external cluster. For instance, if RBAC was not setup correctly on the external cluster, the creation of the root sync will fail to create, but when the remote root sync is to be deleted (rollout resource itself is updated and or removed), the logic to check for the delete the external root sync will fail as well due to the same permissions issue.

Reply:
A note for us: Realized one more edge case where the target cluster never comes back up or is decommissioned without proper cleanup (removing from the targets in the Rollout), the rrs will continue to be hanging in that state. One way to handle that will be to remove all RRS for a cluster when we a cluster is decommissioned (deleting a membership probably).

Reply:
+1 Confirming this check doesn't solve everything but does solve any immediate misconfiguration issues. The ones I came across while developing rollouts were mainly connection, authentication, and authorization. Either the cluster not existing or not having the correct credentials/permissions on the target cluster. The setup for GCPFleet can be involved, more so with the Connect Agent, so it's easy to miss a step.

Reply:
Related comment in the other thread: https://github.com/GoogleContainerTools/kpt/pull/3849/files#r1117840845